sale_delivery_address_modify/models/sale_order.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Opener B.V. (<https://opener.am>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, api
+from openerp.exceptions import Warning as UserError
+from openerp.tools.translate import _
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def write(self, values):
+        """ Propagate a modified delivery address to the procurement group,
+        the pickings and stock moves.
+        """
+        res = super(SaleOrder, self).write(values)
+        if 'partner_shipping_id' in values:
+            for order in self:
+                if not order.procurement_group_id or order.shipped:
+                    continue
+                for procurement in self.env['procurement.order'].search(
```

Comment:
Tested it, but it does not work in the following situation: create a sale order and confirm. Update the delivery address. This updates the delivery address on the picking. Cancel the picking. On the sale order, recreate the picking. The picking has the old delivery address because it is taken from the procurement, not the sale order.